### PR TITLE
[WIP] skeleton slurm license sync script

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/administration/slurm/scancel.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2013-2022 Ghent University
```
Comment (itkovian) [inline lib/vsc/administration/slurm/scancel.py:2]:
```suggestion
# Copyright 2022-2022 Ghent University
```

File: lib/vsc/administration/slurm/scontrol.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+"""
+scontrol commands
+"""
+import logging
+import re
+import shlex
+
+from enum import Enum
+
+from vsc.accountpage.wrappers import mkNamedTupleInstance
```
Comment (itkovian) [inline lib/vsc/administration/slurm/scontrol.py:24]: We should really move this to a more generally available lib.

**Comment by itkovian** [comment]
This is going to break a number of things I think, so before this gets into production, we'll need to address everything that depends on it.

File: test/slurm_sacctmgr.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+#
+# Copyright 2015-2022 Ghent University
```
Comment (itkovian) [inline test/slurm_sacctmgr.py:2]:
```suggestion
# Copyright 2022-2022 Ghent University
```

File: test/slurm_sacctmgr.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-administration
+#
+# All rights reserved.
+#
+"""
+Tests for vsc.administration.slurm.*
+
+@author: Andy Georges (Ghent University)
```
Comment (itkovian) [inline test/slurm_sacctmgr.py:18]:
```suggestion
@author: Stijn Deweirdt (Ghent University)
```

File: lib/vsc/administration/slurm/scontrol.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+LICENSE_RESERVATION_PREFIX = 'external_license_'
+
+class ScontrolTypes(Enum):
+    reservation = "reservation"
+    license = "license"
+    config = "config"
+    partition = "partition"
+
+
+ScontrolReservationFields = [
```
Comment (itkovian) [inline lib/vsc/administration/slurm/scontrol.py:43]: Ideally, these things are versioned. They might change and we should then only add the new ones and use the correct version key.

File: lib/vsc/administration/slurm/sacctmgr.py [hunk 6]
```diff
@@ -0,0 +1,516 @@
+#
+# Copyright 2013-2022 Ghent University
```
Comment (itkovian) [inline lib/vsc/administration/slurm/sacctmgr.py:2]:
```suggestion
# Copyright 2022-2022 Ghent University
```

File: .gitignore [hunk 7]
```diff
@@ -38,3 +38,5 @@ test-reports
 
 
 *.orig
+
+setup.cfg
```
Comment (wdpypere) [inline .gitignore:42]: newline

**Comment by wdpypere** [comment]
unused import sys in setup.py

**Pull request merged**